### PR TITLE
chore: add refactored gh-pages deployments

### PR DESCRIPTION
**Pull request by christopherferreira9**
## Explanation
This PR updates the example apps and adds a refactored gh-pages deployment to allow the deployment of all example dapps of:
- Main
- Latest prod SDK version
- specific branches

It currently supports 3 trigger types:
1. a branch is merged to main
2. manual trigger
3. by applying the `deploy` label on a PR

**Root Url:**
https://metamask.github.io/metamask-sdk

**Explanation:**
The root url contains deployments for:
- prod (updated with the manual trigger)
- main
- each individual PR that had the label applied

**Example:**

https://github.com/user-attachments/assets/7aa42848-7279-4787-8206-7f9c3a4f01c3

**Current limitations and next steps:**
1. Because we want to keep the old deployments and keep the history clean we need to bump the deployment gh action to v4 see [here](https://github.com/peaceiris/actions-gh-pages/pull/456). This is currently being approved by techOps. Once that's approved a follow up bump will be created. This means that only one deployment can be kept.
2. To deploy different custom combinations like latest from `wagmi` + custom `mmsdk` version a new PR needs to be opened for testing, it can then be deleted later
3. A cleanup job needs to be created to delete the older 5-10 builds to avoid having too many artifacts as GH will block that at some point.

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/examples/create-react-app/package.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
   "name": "mm-sdk-cra",
   "version": "1.0.0",
   "private": true,
-  "homepage": "./",
+  "homepage": "https://metamask.github.io/metamask-sdk",
```
Comment (abretonc7s) [inline packages/examples/create-react-app/package.json:5]: the problem if you do that it wont work locally so it needs to be environmentalize

File: packages/examples/create-react-app/sdk-copy.sh [hunk 2]
```diff
@@ -4,10 +4,10 @@
 reldir="$( dirname -- "$0"; )";
 cd "$reldir/../../..";
 SDK_WORKSPACE_DIR="$( pwd; )";
-COMM_LAYER_DIR="$SDK_WORKSPACE_DIR/packages/sdk-communication-layer/dist"
```
Comment (abretonc7s) [inline packages/examples/create-react-app/sdk-copy.sh:7]: why removing the hack?

File: packages/examples/react-metamask-button/yarn.lock [hunk 3]
```diff
@@ -2939,9 +2939,9 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@metamask/sdk-communication-layer@npm:0.28.2":
-  version: 0.28.2
-  resolution: "@metamask/sdk-communication-layer@npm:0.28.2"
+"@metamask/sdk-communication-layer@npm:0.29.2":
```
Comment (abretonc7s) [inline packages/examples/react-metamask-button/yarn.lock:2942]: no 0.30.0 ?

**Review by abretonc7s — CHANGES_REQUESTED** [review]
Not sure about changing the sdk-copy.sh?

File: packages/examples/create-react-app/package.json (hunk 1 above)
Comment (christopherferreira9) [reply packages/examples/create-react-app/package.json:5]: It works locally but doesn't solve the CI part, I'm completely removing it until I find a proper solution.

File: packages/examples/create-react-app/sdk-copy.sh (hunk 2 above)
Comment (christopherferreira9) [reply packages/examples/create-react-app/sdk-copy.sh:7]: see: https://github.com/MetaMask/metamask-sdk/pull/1092/files#diff-86a379f8f6064ca100e36683ec5ef0802fc31797e48023ab83a16d9fe91d86b3R26-R27

File: packages/examples/react-demo/src/SignHelpers.ts [hunk 4]
```diff
@@ -79,7 +79,7 @@ export const send_eth_signTypedData_v4 = async (provider: SDKProvider, chainId:
     console.debug(`ethRequest ${method}`, JSON.stringify(params, null, 4));
     console.debug(`sign params`, params);
     return await provider?.request({ method, params });
-  } catch (e) {
+  } catch (e: any) {
```
Comment (abretonc7s) [inline packages/examples/react-demo/src/SignHelpers.ts:82]: I think using unknown might be better, we want to ban any from the monorepo

File: packages/examples/react-demo/src/SignHelpers.ts [hunk 5]
```diff
@@ -91,13 +91,13 @@ export const send_personal_sign = async (provider: SDKProvider) => {
...
 
-    const sign = await window.ethereum.request({
+    const sign = await window.ethereum!.request({
       method: 'personal_sign',
       params: [hexMessage, from, 'Example password'],
     });
     console.log(`sign: ${sign}`);
     return sign;
-  } catch (err) {
+  } catch (err: any) {
```
Comment (abretonc7s) [inline packages/examples/react-demo/src/SignHelpers.ts:100]: same

File: packages/examples/react-demo/src/SignHelpers.ts [hunk 6]
```diff
@@ -91,13 +91,13 @@ export const send_personal_sign = async (provider: SDKProvider) => {
     const message = 'Hello World from the Create React dapp!';
     const hexMessage = '0x' + Buffer.from(message, 'utf8').toString('hex');
 
-    const sign = await window.ethereum.request({
+    const sign = await window.ethereum!.request({
```
Comment (abretonc7s) [inline packages/examples/react-demo/src/SignHelpers.ts:94]: in this case also might be better to check before and print error if window.ethereum not present but overall might be better to have example using the provider

**Review by abretonc7s — CHANGES_REQUESTED** [review]
tiny changes that we should take care as we update the examples

File: packages/examples/react-demo/src/SignHelpers.ts (hunk 5 above)
Comment (christopherferreira9) [reply packages/examples/react-demo/src/SignHelpers.ts]: fixed

File: packages/examples/react-demo/src/SignHelpers.ts (hunk 6 above)
Comment (christopherferreira9) [reply packages/examples/react-demo/src/SignHelpers.ts]: seems like it was using the old `window.ethereum` approach. Switched to the same as sign_v4

**Review by abretonc7s — APPROVED** [review]

**Comment by christopherferreira9** [comment]
@SocketSecurity ignore-all

**Pull request merged**